x/oracle/types/keys.go

```diff
@@ -36,8 +36,7 @@ var (
 // GetExchangeRateKey - stored by *denom*
 func GetExchangeRateKey(denom string) (key []byte) {
 	key = append(key, KeyPrefixExchangeRate...)
-	key = append(key, []byte(denom)...)
-	return append(key, 0) // append 0 for null-termination
+	return append(key, []byte(denom)...)
```

Comment:
Nice catch. Does this address (1) from #305?

Reply:
Wait, on second thought I'm not confident this is the correct approach. I'm pretty sure we _want_ the null-terminator, so we can iterate effectively. Instead, I think we need to remove the null-terminator in the query method or whatever does the iterating/object retrieval.